tests/neg/17579.check

```diff
@@ -0,0 +1,26 @@
+-- [E199] Syntax Error: tests/neg/17579.scala:4:10 ---------------------------------------------------------------------
+4 |    final val v1 = 42 // error
+  |          ^^^
+  |          The final modifier is not allowed on local definitions
+-- [E199] Syntax Error: tests/neg/17579.scala:5:15 ---------------------------------------------------------------------
+5 |    final lazy val v2 = 42 // error
+  |               ^^^
+  |               The final modifier is not allowed on local definitions
+-- [E088] Syntax Error: tests/neg/17579.scala:6:10 ---------------------------------------------------------------------
+6 |    final private val v3 = 42 // error
+  |          ^^^^^^^
+  |          Expected start of definition
```

Comment:
Why a different error message here?

Reply:
I see why: since `private` is also not a local modifier token.

Reply:
This means we could generalize the solution. We could always parse all tokens and then flag those that are not legal at this point. But this could be another PR.